godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -575,17 +575,44 @@ macro_rules! impl_float_vector_fns {
                 )
             }
 
-            /// Returns the vector scaled to unit length. Equivalent to `self / self.length()`. See
-            /// also `is_normalized()`.
+            /// Returns the vector scaled to unit length or [`None`], if called on a zero vector.
+            ///
+            /// Equivalent to `self / self.length()`. See also [`normalized()`][Self::normalized] and [`is_normalized()`][Self::is_normalized].
+            #[inline]
+            pub fn try_normalized(self) -> Option<Self> {
+                if self == Self::ZERO {
+                    return None;
+                }
+
+                // Copy Godot's implementation since it's faster than using glam's normalize_or_zero().
+                Some(self / self.length())
+            }
+
+            /// Returns the vector scaled to unit length.
```

Comment:
This could now be prefixed with :warning: symbol, like other panicking methods that have a `try_` overload, e.g. `Gd::cast()`.